### DISCUSSION TITLE
Using LLVM instrumentation

### DISCUSSION THREAD

**Discussion by winksaville** [xd009642/tarpaulin#739] [General]
So I've run into `Misses in coverage` #351 and was wondering if moving to LLVM coverage instrumentation #549 is expected to improve upon the coverage accuracy?

**Comment by xd009642** [comment]
Yes it will as the compiler will be adding instructions into the code to mark what's hit instead of having to rely on correctly placing breakpoints

**Comment by winksaville** [comment]
Where does it rank in your priority list?

**Reply by xd009642** [reply]
It's the top item right now, though a close second is turning the 0.18.0-alpha releases into a proper 0.18.0. Some of the tasks I've done which are mention in the llvm coverage issue are making tarpaulin agnostic to any tracing engine so I can plug in the llvm code when time comes. 

I've got this repo https://github.com/xd009642/llvm-profparser which is where I'm doing that work and currently I can parse the profile data formats and I'm now working on taking the binaries and the parsed profiling data so I can map the coverage counters to source lines. This is all to make it so users aren't required to use rustup or install additional rust CLI tools to use tarpaulin with llvm coverage 

I'm also currently away on a business trip so things have been moving a bit slower but I'll be back home in a week or so and should get more of my free time back :grin:

**Comment by winksaville** [comment]
SG.

Is the plan to include LLVM instrumentation in 0.18.0 release?

I'm guessing not, as I'm assuming it's still a ways out.

**Reply by xd009642** [reply]
The plan is 0.19.0 just because there's a lot of changes in 0.18.0 I'd like to get properly out there before adding even more to the release

**Reply by winksaville** [reply]
Good plan, as I need the --follow- exec in 0.18.0 :)
